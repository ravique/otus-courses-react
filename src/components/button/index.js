import React, {Component} from "react";
import {Link} from 'react-router-dom';

const classNames = require('classnames');

export default class Button extends Component {

    constructor() {
        super();
        this.state = {
            pressed: false
        };
    }

    handleClick = () => {
         this.setState(prevState => ({
            pressed: !prevState.pressed
        }));
    };

    render() {

        const {url, text} = this.props;

        let btnClass = classNames({
            'button': true,
            'button_pressed': this.state.pressed
        });

        return (
            <Link to={url} className={btnClass} onClick={this.handleClick}>{text}</Link>
        )
    }
}

