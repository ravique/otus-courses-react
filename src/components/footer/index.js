import React from "react";

const Footer = () => (
    <div className="footer">
        <span>I am the footer!</span>
    </div>
);


export default Footer
