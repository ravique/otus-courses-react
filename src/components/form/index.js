import React, {Component} from 'react';
import PropTypes from 'prop-types';

import {capitalizeFirstLetter} from "../../utils/utils";
const classNames = require('classnames');

class FormLabel extends Component {
    render() {
        return (
            <label htmlFor={this.props.name} className="form__label">
                {this.props.text ? this.props.text : capitalizeFirstLetter(this.props.name)}
            </label>
        )
    }
}

class FormInput extends Component {

    handleChange = (event) => {
        this.props.eventHandler(event);
    };

    render() {
        const inputCls = classNames({
            'form__input': true,
            'form__input_error': this.props.error
        });

        return (
            <input
                type={this.props.type}
                className={inputCls}
                name={this.props.name}
                placeholder={this.props.placeholder}
                value={this.props.value}
                onChange={this.handleChange}
                accept={this.props.accept}
            />
        )
    }
}

FormInput.propTypes = {
    type: PropTypes.string,
    name: PropTypes.string,
    placeholder: PropTypes.string,
    value: PropTypes.string,
    accept: PropTypes.string
};

export {FormLabel, FormInput};