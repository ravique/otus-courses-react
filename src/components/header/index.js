import React from "react";
import {Logo, SmartMenu} from '../menu/menu'

const Header = () => (
    <div className='header'>
        <Logo/>
        <SmartMenu/>
    </div>
);

export default Header;