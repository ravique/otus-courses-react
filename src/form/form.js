import React, {Component} from 'react';
import {capitalizeFirstLetter} from "../utils/utils";
const classNames = require('classnames');

class FormLabel extends Component {
    render() {
        return (
            <label htmlFor={this.props.name} className="form__label">
                {this.props.text ? this.props.text : capitalizeFirstLetter(this.props.name)}
            </label>
        )
    }
}

class FormInput extends Component {

    handleChange(event) {
        this.props.eventHandler(event);
    }

    render() {
        const inputCls = classNames({
            'form__input': true,
            'form__input_error': this.props.error
        });

        return (
            <input
                type={this.props.type}
                className={inputCls}
                name={this.props.name}
                placeholder={this.props.placeholder}
                value={this.props.value}
                onChange={this.handleChange.bind(this)}
            />
        )
    }
}

export {FormLabel, FormInput};