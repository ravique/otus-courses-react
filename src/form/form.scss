.form {
  width: 500px;
  margin: 0 auto;

  &__input {
    width: 100%;
    padding: 10px;
    border: 1px solid $white;
  }

  &__label {
    display: block;
    margin: 15px 0 0 0;

    &_errrors {
      color: $red;
    }

  }

  &__errors {
    color: $red;

    &__message {
      display: block;
      margin: 5px 0;
    }
  }
}

