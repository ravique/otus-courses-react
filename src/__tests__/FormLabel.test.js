import React from 'react';
import {shallow} from 'enzyme';
import {shallowToJson} from 'enzyme-to-json';
import '../../jest.config'
import {FormLabel} from "../components/form";

describe('FormLabel', () => {

    let output;

    beforeEach(() => {
        output = shallow(
            <FormLabel name='test'/>
        );
    });

    it('should render form label', () => expect(shallowToJson(output)).toMatchSnapshot());
    it('should render text of label', () => expect(output.text()).toContain('Test'));
});
