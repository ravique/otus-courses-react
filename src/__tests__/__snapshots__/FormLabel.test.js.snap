// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormLabel should render form label 1`] = `
<label
  className="form__label"
  htmlFor="test"
>
  Test
</label>
`;
